### ISSUE TITLE
Clean breaks browser sync-ui 

### ISSUE THREAD

**Issue by thomsa** [yarnpkg/yarn#1396]
**Do you want to request a *feature* or report a *bug*?**
BUG
**What is the current behavior?**
yarn clean removes history/history.plugin from browser-sync-ui 
```
Error: Cannot find module './plugins/history/history.plugin'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.Module._load (module.js:388:25)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (~\node_modules\browser-sync-ui\lib\UI.js:13:25)
    at Module._compile (module.js:541:32)
    at Module._extensions..js (module.js:550:10)
    at Object.require.extensions.(anonymous function) [as .js] (~\node_modules\babel-register\lib\node.js:152:7)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (~\node_modules\browser-sync-ui\index.js:3:19)
    at Module._compile (module.js:541:32)
    at Module._extensions..js (module.js:550:10)
    at Object.require.extensions.(anonymous function) [as .js] (~\node_modules\babel-register\lib\node.js:152:7)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (~\node_modules\browser-sync\lib\browser-sync.js:26:22)
    at Module._compile (module.js:541:32)
    at Module._extensions..js (module.js:550:10)
    at Object.require.extensions.(anonymous function) [as .js] (~\node_modules\babel-register\lib\node.js:152:7)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
```

**If the current behavior is a bug, please provide the steps to reproduce.**
1. Have browser-sync as dependency
2. Install dependencies
3. Run yarn clean
4. Try to gulp serve

**What is the expected behavior?**
Clean should not remove the necessary files.

**Please mention your node.js, yarn and operating system version.**
System: Windows 10
NodeJS: V6.2.1
Yarn: 0.16.1

**Comment by Pixelbird** [comment]
Can confrim, history folder missing from node_modules/browser-sync-ui/lib/plugins
```
Error: Cannot find module './plugins/history/history.plugin'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (C:\path\to\project\node_modules\browser-sync-ui\lib\UI.js:13:25)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
```

**Comment by klucznik** [comment]
Same on MacOS, yarn 0.16.1

**Comment by thomsa** [comment]
Unfortunately that wasn't the only file. Clean is really aggressive.

**Comment by cezarlz** [comment]
Same here!

**Comment by marzepani** [comment]
Add this exclude filter to your `.yarnclean` file and call `yarn add browsersync`again:
```terminal
!browser-sync-ui/lib/plugins/history
```

([.yarnclean Gist](https://gist.github.com/marzepani/4e4f3334960945722eca458fe112c21a))

**Comment by kittens** [comment]
Fixed via #1430.

**Comment by ibbatta** [comment]
Same here... actually still brokes
